src/shared/utilities/vsCodeUtils.ts

```diff
@@ -47,6 +48,49 @@ export async function closeAllEditors() {
     }
 }
 
+/**
+ * Gives context regarding the user + aws extension.
+ */
+export class ExtensionUse {
+    private isFirstUseState: boolean | undefined = undefined
+
+    static isExtensionFirstUseKey = 'isExtensionFirstUse'
+
+    /**
+     * IMPORTANT: This must be run first.
+     *
+     * Updates the {@link isFirstUse} variable to indicate
+     * if this is the first use of the extension.
+     */
+    updateIsFirstUse(ctx: vscode.ExtensionContext): void {
+        const isFirst = ctx.globalState.get(ExtensionUse.isExtensionFirstUseKey, true)
+
+        if (isFirst) {
+            ctx.globalState.update(ExtensionUse.isExtensionFirstUseKey, false)
+        }
+
+        this.isFirstUseState = isFirst
+    }
```

Comment:
Should we use `globals` to do this implicitly?

Reply:
Makes things much simpler using this. Updating to not need to use a class at all.